ocaml/xapi/server_helpers.ml

```diff
@@ -118,26 +128,44 @@ let dispatch_exn_wrapper f =
   with exn -> let code, params = ExnHelper.error_of_exn exn in API.response_of_failure code params
 
 let do_dispatch ?session_id ?forward_op ?self called_async supports_async called_fn_name op_fn
-    marshaller_fn fd http_req label generate_task_for =
+    marshaller_fn fd http_req label sync_ty generate_task_for =
```

Comment:
The `called_async` arg is now obsolete, because it's just the same as ``sync_ty <> `Sync``. Let's remove it, and then also the `__async` value can be removed from server.ml and the helper above.